### PR TITLE
HW_4: first attempt

### PR DESCRIPTION
**Pull request by alchemixXx**

### CODE REVIEW

File: build_in_methods_iterators/homework.py [hunk 1]
```diff
@@ -14,7 +16,15 @@ def task_1_fix_names_start_letter(data: DT) -> DT:
         fix_names_start_letters([{'name': 'Alex', 'age': 26}, {'name': 'denys', 'age': 89}])
         >>> [{'name': 'Alex', 'age': 26}, {'name': 'Denys', 'age': 89}]
     """
-    pass
+    for member in data:
+        if 'name' in member:
+            member["name"] = member["name"].capitalize()
+    return data
```
Comment (dgorbachevskyy) [inline build_in_methods_iterators/homework.py]: use dictionary method update

Comment (dimigor) [inline build_in_methods_iterators/homework.py]: use in list comprehension dictionary method update

File: build_in_methods_iterators/homework.py [hunk 2]
```diff
@@ -35,44 +51,61 @@ def task_3_find_item_via_value(data: DT, value) -> DT:
...
-    pass
+    # return [str(x) for x in data if len(str(x)) == min(map(lambda x: len(str(x)), data), default=None)][0]
 
+    # created_dict = {str(item):len(str(item)) for item in data}
+    # try:
+    #     return list(created_dict.keys())[list(created_dict.values()).index(min(created_dict.values(), default=None))]
+    # except:
+    #     return None
+    try:
+        return [str(key) for (key, value) in {item: len(str(item)) for item in data}.items() if
```
Comment (dimigor) [inline build_in_methods_iterators/homework.py]: use simple loop

**Review by dimigor — CHANGES_REQUESTED** [review]

File: build_in_methods_iterators/homework.py (hunk 2 above)
Comment (alchemixXx) [reply build_in_methods_iterators/homework.py]: done

File: build_in_methods_iterators/homework.py (hunk 1 above)
Comment (alchemixXx) [reply build_in_methods_iterators/homework.py]: done!

**Review by dimigor — APPROVED** [review]

**Pull request merged**